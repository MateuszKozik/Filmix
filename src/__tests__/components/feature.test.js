import { render } from "@testing-library/react";
import { Feature } from "../../components";

describe("<Feature />", () => {
	it("renders the <Feature /> with populated data", () => {
		const { container, getByText } = render(
			<Feature>
				<Feature.Title>Unlimited films and series.</Feature.Title>
				<Feature.SubTitle>Join us Now!</Feature.SubTitle>
			</Feature>
		);

		expect(getByText("Unlimited films and series.")).toBeTruthy();
		expect(getByText("Join us Now!")).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});

	it("renders the <Feature /> with just a title", () => {
		const { container, getByText, queryByText } = render(
			<Feature>
				<Feature.Title>Unlimited films and series.</Feature.Title>
			</Feature>
		);

		expect(getByText("Unlimited films and series.")).toBeTruthy();
		expect(queryByText("Join us Now!")).toBeFalsy();
		expect(container.firstChild).toMatchSnapshot();
	});

	it("renders the <Feature /> with just a subtitle", () => {
		const { container, getByText, queryByText } = render(
			<Feature>
				<Feature.SubTitle>Join us Now!</Feature.SubTitle>
			</Feature>
		);

		expect(queryByText("Unlimited films and series.")).toBeFalsy();
		expect(getByText("Join us Now!")).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
