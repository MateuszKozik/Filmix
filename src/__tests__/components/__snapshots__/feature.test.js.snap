// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders the <Feature /> with just a subtitle 1`] = `
<div
  class="feature__Container-sc-17uyeqc-0 bvAA-dh"
>
   
  <h2
    class="feature__SubTitle-sc-17uyeqc-2 kqjfyS"
  >
    Join us Now!
  </h2>
</div>
`;

exports[`<Feature /> renders the <Feature /> with just a title 1`] = `
<div
  class="feature__Container-sc-17uyeqc-0 bvAA-dh"
>
   
  <h1
    class="feature__Title-sc-17uyeqc-1 SngDu"
  >
     
    Unlimited films and series.
     
  </h1>
</div>
`;

exports[`<Feature /> renders the <Feature /> with populated data 1`] = `
<div
  class="feature__Container-sc-17uyeqc-0 bvAA-dh"
>
   
  <h1
    class="feature__Title-sc-17uyeqc-1 SngDu"
  >
     
    Unlimited films and series.
     
  </h1>
  <h2
    class="feature__SubTitle-sc-17uyeqc-2 kqjfyS"
  >
    Join us Now!
  </h2>
</div>
`;
